src/illuminatio/illuminatio_runner.py

```diff
@@ -153,92 +176,59 @@ def run_tests_for_target(network_ns, ports, target):
-        nmap_cmd = ["nmap", "-oX", result_file.name, "-Pn", "-p", port_string, svc_ip]
-        LOGGER.info("running nmap with cmd %s", nmap_cmd)
-        prc = None
-        with Namespace(network_ns, "net"):
-            prc = subprocess.run(
-                nmap_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-            )
-        if prc is None or prc.returncode:
-            LOGGER.error("Executing nmap in foreign net ns failed! output:")
-            LOGGER.error(prc.stderr)
-            LOGGER.debug(prc)
-            return {
-                port_string: {
-                    "success": False,
-                    "error": "Couldn't nmap host %s with hostname %s"
-                    % (target, svc_dns_entry),
-                }
-            }
 
-        LOGGER.info("finished running nmap")
-        LOGGER.debug("Error log: %s", prc.stderr)
-        # when not using shell the output from nmap contains \\n instead of newline characters
-        LOGGER.info("Stdout: %s", str(prc.stdout).split("\\n"))
-        return extract_results_from_nmap_xml_file(result_file, port_on_nums, target)
+    ipv6_arg = ""
+    if ipaddress.ip_address(svc_ip).version == 6:
+        ipv6_arg = "-6"
 
+    # nmap that target TODO: handle None ip
+    nm_scanner = nmap.PortScanner()
+    with Namespace(network_ns, "net"):
+        nm_scanner.scan(svc_ip, arguments=f"-n -Pn -p {port_string} {ipv6_arg}")
+    LOGGER.info("Ran nmap with cmd %s", nm_scanner.command_line())
 
-def pod_list_contains_pod(pod, pod_list):
-    """
-    Checks whether a list of pods contains a given pod
-    """
-    LOGGER.debug("Searching for pod %s", pod)
-    if isinstance(pod, ConcreteClusterHost):
-        is_on_node = any([pod == pod_on_node for pod_on_node in pod_list])
-        if is_on_node:
-            LOGGER.debug("Pod %s in namespace %s was found", pod.name, pod.namespace)
-        else:
-            LOGGER.debug(
-                "Pod %s in namespace %s isn't on this node", pod.name, pod.namespace
-            )
-        return is_on_node
-
-    LOGGER.error("Found non-ConcreteClusterHost host in cases: %s", pod)
-    return False
+    return extract_results_from_nmap(nm_scanner, port_on_nums, target)
 
 
-def extract_results_from_nmap_xml_file(result_file, port_on_nums, target):
+def extract_results_from_nmap(nmap_res, port_on_nums, target):
     """
-    Extracts the results of an nmap scan from an xml result file into a dictionary
+    Extracts the results of an nmap scan into a dictionary
     """
-    xml = ElementTree.parse(result_file.name)
-    hosts = [h for h in xml.getroot().iter("host")]
+    hosts = nmap_res.all_hosts()
     if len(hosts) != 1:
         LOGGER.error(
-            "Fund %s a single host in nmap results but expected only one target to be probed",
+            "Found %s a single host in nmap results but expected only one target to be probed",
```

Comment:
Not part of your change, but: Does this error message make sense? What is replaced for `%s` in that string?

Reply:
The number of hosts e.g. `0` if we didn't find one or multiple hosts (which shouldn't be the case). Actually I would also tend to remove it since more or less the same message is part of the result.